src/components/forms/RangeInput/RangeInput.tsx

```diff
@@ -17,47 +19,38 @@ type RangeInputProps = {
 export const RangeInput = ({
   className,
   inputRef,
+  textPreposition,
+  textUnit,
   ...inputProps
 }: RangeInputProps & JSX.IntrinsicElements['input']): React.ReactElement => {
   const classes = classnames('usa-range', className)
   // input range defaults to min = 0, max = 100, step = 1, and value = (max/2) if not specified.
   const defaultMin = 0
   const defaultMax = 100
-  const { min, max, defaultValue } = inputProps
+  const defaultVal = Math.round(defaultMax / 2)
+  const defaultUnit = ''
+  const defaultPreposition = 'of'
+  const { min, max } = inputProps
   const rangeMin = min || defaultMin
   const rangeMax = max || defaultMax
-  const ariaMin = inputProps['aria-valuemin'] || rangeMin
-  const ariaMax = inputProps['aria-valuemax'] || rangeMax
-  const calculatedAriaValueNow =
-    inputProps['aria-valuenow'] ||
-    defaultValue ||
-    (rangeMax < rangeMin ? rangeMin : rangeMin + (rangeMax - rangeMin) / 2)
-  const convertValueType = (
-    value: string | number | readonly string[]
-  ): number | undefined => {
-    if (typeof value === 'number' || typeof value === 'string') {
-      return Number(value)
-    }
-    return undefined
-  }
-  const [ariaValue, setAriaValue] = React.useState<number | undefined>(
-    convertValueType(calculatedAriaValueNow)
-  )
-  const onValueChange: React.ChangeEventHandler<HTMLInputElement> = (e) => {
-    if (!inputProps['aria-valuenow']) setAriaValue(e.target.valueAsNumber)
-  }
+  const rangeUnit = textUnit || defaultUnit
+  const rangePreposition = textPreposition || defaultPreposition
+  const rangeValue = inputProps.defaultValue || defaultVal
+  const [value, setValue] = useState(rangeValue)
+  const callout = `${value} ${rangeUnit} ${rangePreposition} ${rangeMax}`
 
   return (
     <input
       data-testid="range"
+      aria-valuetext={callout}
       className={classes}
       ref={inputRef}
       type="range"
       {...inputProps}
-      aria-valuemin={ariaMin}
-      aria-valuemax={ariaMax}
-      aria-valuenow={ariaValue}
-      onChange={(e): void => onValueChange(e)}
+      min={rangeMin}
+      max={rangeMax}
+      defaultValue={value}
```

Comment:
I think this should be:

```suggestion
      value={value}
```
React wants only a controlled or uncontrolled input. In our case, we're using a react state to control the value, so its a controlled input, so we should be using the `value` prop